forta-bots/DAI-Price-Monitor/src/agent.ts

```diff
@@ -0,0 +1,37 @@
+import BigNumber from "bignumber.js";
+import { providers } from "ethers";
+import { Finding, getEthersProvider, HandleTransaction, Initialize, TransactionEvent } from "forta-agent";
+import Fetcher from "./fetcher";
+import NetworkData, { NETWORK_MAP } from "./network";
+import NetworkManager from "./network";
+import { SPREAD_THRESHOLD } from "./utils";
+import provideCurveHandleTransaction from "./curve.event";
+import provideUniswapHandleTransaction from "./uniswap.event";
+
+const networkManager: NetworkData = new NetworkManager(NETWORK_MAP);
+
+export const initialize =
+  (provider: providers.Provider): Initialize =>
+  async () => {
+    const { chainId } = await provider.getNetwork();
+    networkManager.setNetwork(chainId);
+  };
+
+export const provideBotHandleTransaction = (
+  data: NetworkData,
+  fetcher: Fetcher,
+  spreadThreshold: BigNumber
+): HandleTransaction => {
+  const uniswapEventHandler: HandleTransaction = provideUniswapHandleTransaction(data, fetcher, spreadThreshold);
+  const curveEventHandler: HandleTransaction = provideCurveHandleTransaction(data, spreadThreshold);
+  return async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    let findings: Finding[] = [];
+    findings = [...(await uniswapEventHandler(txEvent)), ...(await curveEventHandler(txEvent))];
+    return findings;
+  };
```

Comment:
I would suggest:
```suggestion
  const handlers: HandleTransaction = [
    provideUniswapHandleTransaction(data, fetcher, spreadThreshold),
    provideCurveHandleTransaction(data, spreadThreshold),
  ];
  return async (txEvent: TransactionEvent): Promise<Finding[]> => 
    (await Promise.all(handlers.map(handler => handler(txEvent))).flat();
```
So in the future you only need to register new handlers in the `handlers` array. This also adds Promise.all usage.